src/twisted/test/test_defer.py

```diff
@@ -3566,7 +3566,7 @@ def test_deprecatedTimeout(self) -> None:
         """
         L{twisted.internet.defer.timeout} is deprecated.
         """
-        deferred: Deferred[None] = Deferred()
+        deferred: Deferred[object] = Deferred()
         defer.timeout(deferred)
```

Comment:
defer.timeout should probably take a `Deferred[None]` rather than a `Deferred[object]`

Reply:
It's deprecated anyway so it doesn't really matter but `defer.timeout` calls `deferred.callback(None)` not `deferred.callback(object)`